mlir/test/fusion/e2e/generate-fusion-tests.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+
+import os
+import argparse
+import shutil
+import glob
+import itertools
+import tomli
+
+RANDTYPE = {'f32' : 'float',
+            'f16' : 'float',
+            'bf16' : 'float',
+            'i32' : 'int',
+            'i8' : 'int'}
+
+def generate_option_list(table, key1, key2):
+    options_list=[]
+    for item in table[key1]:
+        options=[]
+        for value in item[key2]:
+            options.append(value)
+        options_list.append(options)
+    combinations=[]
+    for opt in itertools.product(*options_list):
+        combinations.append(opt);
+    return combinations
+
+def generate_tosa_to_rock_test(indir, outdir, type, file, opspec):
+    opname,op = opspec
+    with open(f"{indir}/{file}.e2e.template") as f:
+        template = f.read()
+    outfile = f"{outdir}/{file}-{opname}-{type}.e2e.mlir"
+    with open(outfile, 'w') as f:
+        # The instruction operand is in the middle for clamp, so we can't simply
+        # substitute.  With a regexp replace we could capture the actual operand
+        # but for now we'll just "know" what we have.
+        op = op.format(operand='(%0)')
+        f.write(template.format(op=op, type=type,
+                                randtype=RANDTYPE[type],
+                                randkind='fixed' if opname != 'rsqrt' else '1',
+                                # So far clone-verification only works with f32.  Also, tanh
+                                # fails it because math.tanh is converted by gpu-to-rocdl pass
+                                # which isn't run on the cloned function.
+                                disablep='-DISABLE' if type != 'f32' or opname == 'tanh' else ''))
+
+def generate_type_only_test(indir, outdir, type, file):
+    with open(f"{indir}/{file}.e2e.template") as f:
+        template = f.read()
+    outfile = f"{outdir}/{file}-{type}.e2e.mlir"
+    with open(outfile, 'w') as f:
+        f.write(template.format(type=type,
+                                # So far clone-verification only works with f32.
+                                disablep='-DISABLE' if type != 'f32' else ''))
+
+def toml_loop(toml, indir, outdir):
+    for suite in toml['suite']:
+        combinations = generate_option_list(suite, 'axis', 'values')
+        for test in combinations:
+            if suite['name'] == "tosa-to-rock tests":
```

Comment:
Instead of keying off the name field, would you be open to adding a `type` field to `suite`

Reply:
Okay, but I called it "kind" because "type" has so many other uses.